store/local_cache_supplier_roles.go

```diff
@@ -18,10 +18,11 @@ func (s *LocalCacheSupplier) handleClusterInvalidateRole(msg *model.ClusterMessa
 }
 
 func (s *LocalCacheSupplier) RoleSave(ctx context.Context, role *model.Role, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
+	result := s.Next().RoleSave(ctx, role, hints...)
 	if len(role.Id) != 0 {
-		defer s.doInvalidateCacheCluster(s.roleCache, role.Name)
+		s.doInvalidateCacheCluster(s.roleCache, role.Name)
```

Comment:
This changes what happens. Doing the `RoleSave()` call before the `if len(role.Id) != 0` means that condition will always be `true` now because `RoleSave()` sets the ID.

Reply:
Ah, I missed that. I'll revert both of these